### ISSUE TITLE
Ability to modify network response after passthrough

### ISSUE THREAD

**Issue by mohammadalipak** [miragejs/miragejs#579]
Currently, we can either mock the api response using `routes` or use the real network response using `passthrough`. However, we can't modify the real network response before it's piped into the app. It would be really convenient to be able to patch or modify the network response using MirageJs.

**Comment by samselikoff** [comment]
Yes, I think folks have asked about this in the past. Mirage route handlers are promise-aware so you could return a promise that makes the original request, then modify it in the `.then` before returning. But I suppose if you tried to do that now you'd hit a loop or something like that?

Definitely open to an API that would let you make the original request in your route handler. Maybe we could inject an `originalRequest` arg into the route handler that you could call + await, in order to munge it before resolving the handler.

**Comment by robatwilliams** [comment]
What I did for this was capture the real `fetch()` into a variable before calling `createServer()`, then call it from my handler (mapping the fakeRequest's options over to what native fetch expects).

Would be good to have this built-in.
